metaflow/plugins/argo/argo_workflows_cli.py

```diff
@@ -123,7 +124,13 @@ def argo_workflows(obj, name=None):
     "--only-json",
     is_flag=True,
     default=False,
-    help="Only print out JSON sent to Argo Workflows. Do not deploy anything.",
+    help="Only print out JSON sent to Argo Workflows. Do not deploy anything. DEPRECATED, prefer --output-yaml instead",
+)
+@click.option(
+    "--only-yaml",
+    is_flag=True,
+    default=False,
+    help="Only print out yaml sent to Argo Workflows. Do not deploy anything. Includes all objects",
```

Comment:
Unfortunately - this is a bit incorrect. The description for `--only-json` also needs to be fixed (or the implementation updated). Currently, using `--only-json` updates the sensors - which needs to be addressed. 

My preference would be to fix the behavior of `--only-json` so that it is side-effect-free, introduce additional flags to emit other objects (sensors etc.) as json objects so that there is no need for introducing a pyyaml dependency. the output of `--only-json` can then be piped to a local script that can convert the json payload into a yaml.

Reply:
Ack. Let me see if I can fix --only-json in another PR.

As for json vs yaml, see my above comment.

Reply:
@savingoyal is this true? Where do the sensors get updated? I traced the code and couldn't find where this'd happen.